### PR TITLE
add participants count to column in batch overview

### PR DESCRIPTION
**Pull request by phipsae**
Hey Carlos,

I just made a small adjustment by adding the participants count next to the graduates count, so we can get a better sense of how many started and how many completed successfully (in our opinion).

Screenshot 2024-11-05 at 13 42 46

Best, Philip

### CODE REVIEW

File: packages/react-app/pages/admin/batches.jsx [hunk 1]
```diff
@@ -115,15 +117,38 @@ export default function Batches({ serverUrl, userRole, mainnetProvider }) {
...
+          if (builder.batch && builder.batch.number === String(batchName)) {
+            participantsCounts[batchName]++;
+          }
+        });
+      },
+      [batches],
+    );
+
+    setParticipantsCount(participantsCounts);
+  }, [batches, serverUrl]);
```
Comment (carletex) [inline packages/react-app/pages/admin/batches.jsx:139]: Damn, this nested loop is ugly haha

We are also making 3 requests to the server to render this page, which doesn't make sense. 

Let's update the backend as needed so we don't need to do this. We could:

- Extend the /batches endpoint so it comes back with the count calculations
or
- Create a new `/batches/stats` route that includes this data

So we'll only need to do 1 request. Also will be helpful here https://github.com/BuidlGuidl/buidlguidl.com/pull/54#discussion_r1830737984

**Review by carletex — CHANGES_REQUESTED** [review]
Thanks @phipsae !

Added a review comment

**Comment by phipsae** [comment]
Hey Carlos, 

I changed the /batches route so that it gives you the totalParicipants and graduates count for every batch. 
Pls let me know if I should make any adjustments.

Thank you!

**Review by carletex — APPROVED** [review]
This looks great. Thanks @phipsae !!

**Pull request merged**